Adding ULF app

playpaolo opened the pull request:
Latest version of Updated Lagrangian Fluid models including the glass forming module

playpaolo committed:
Updating to macth new format

playpaolo committed:
Cleaning obsolete elements

(those of Guillermo Casas - under his agreement)

playpaolo committed:
Updated the SetUpDofSet function

playpaolo committed:
Added elements and conditions and updated constructors

Added new elements and conds of axisymmetric formulation. Removed custom io (obsolete). Updated builder and solver to new format. And some other minor changes.

playpaolo committed:
python_scripts folder inside the application folder removed

(with a  unique file: beam_profiles.csv)

playpaolo committed:
removed results files

playpaolo committed:
Updating (2to3 Py syntax) and adding files for glass simulation

added a separate trigen for glass.

playpaolo committed:
Fixing a bug in re-meshing with structure

1) rectified the alpha-shape cases (removing the obsolete ones) in trigen_glass_forming
2) in ulf_PGLASS.py added functions that assign nodal_h to the lonely nodes

playpaolo committed:
2to3

corrected tabs in ulf_PGLASS

playpaolo committed:
Nodal h at lonely nodes

added lines that prescribe nodal_h at lonely nodes

playpaolo committed:
Meshing parameteres

simply changed the mesh quality criteria

playpaolo committed:
reform dof set at each iteration enabled

playpaolo committed:
minor change

playpaolo committed:
updated glass files

playpaolo committed:
gitignore

playpaolo committed:
merging master into feature-ulf

playpaolo committed:
added missing argument 

to ResizeAndInitializeVectors(self.scheme, self.pA, self.pDx, self.pb, self.model_part.Elements, self.model_part.Conditions, self.model_part.ProcessInfo);

(self.scheme) was missing

loumalouomega commented on .gitignore:
```diff
@@ -55,3 +55,15 @@ kratos/includes/kratos_version.h
 *.mesh
 *.o.sol
 *.o.mesh
+
```
This should be locally in your .git/info/exclude

See https://github.com/KratosMultiphysics/Kratos/wiki/How-to-make-Git-ignore-the-files-resulting-from-a-compilation-without-conflicts-in-.gitignore

loumalouomega commented on applications/MeshingApplication/custom_python/add_meshers_to_python.cpp:
```diff
@@ -118,6 +119,19 @@ void TriRegenerateMesh(TriGenPFEMModeler& Mesher, char* ElementName, char* Condi
+//											//
+//////////////////////////////////////////////////////////////////////////////////////////
+
+void TriRegenerateMeshGLASS(TriGenGLASSModeler& Mesher, char* ElementName, char* ConditionName, ModelPart& model_part,NodeEraseProcess& node_erase, bool rem_nodes, bool add_nodes, double alpha_shape, double h_factor )
```
Just a comment, the input variables in a method should be in CammelCase https://github.com/KratosMultiphysics/Kratos/wiki/Style-Guide

playpaolo replied:
didnt get it. could you please indicate which of the variables doesnt follow this format?

pooyan-dadvand replied:
`model_part` -> `rModelPart`
`node_erase` -> `TheNodeEraseProcess`
and so on

playpaolo replied:
okay, cool! thanks!

loumalouomega commented on applications/ULFapplication/ULF_application.cpp:
```diff
@@ -1,7 +1,7 @@
 //
 //   Project Name:        Kratos
 
-//   Last Modified by:    $Author: virginia $
```
The new headers are like:

~~~c
//    |  /           |
//    ' /   __| _` | __|  _ \   __|
//    . \  |   (   | |   (   |\__ \.
//   _|\_\_|  \__,_|\__|\___/ ____/
//                   Multi-Physics
//
//  License:		 BSD License
//					 Kratos default license: kratos/license.txt
//
//  Main authors:    Riccardo Rossi
//  Colaborator:     Vicente Mataix Ferrandiz
//
~~~

playpaolo replied:
thank you Vicente. Is there any way of making this change automatically or do I have to do it by manual editing?

loumalouomega replied:
Unfortunately you should do this by hand, at least what I know

pooyan-dadvand replied:
@playpaolo it is encouraged to do it but you can postponed it to next commits if you like.

pooyan-dadvand replied:
Just to add that right now is more important to have a compilable ulf app as soon as possible.

playpaolo replied:
Okay, so I will start by correcting the rest of things

pooyan-dadvand commented on applications/MeshingApplication/external_includes/tetgen_pfem_refine_GLASS_CORRECT.h:
```diff
@@ -0,0 +1,1792 @@
+					);
+
+
+#ifdef _DEBUG
```
I would suggest to use KRATOS_DEBUG instead.

pooyan-dadvand commented on applications/MeshingApplication/external_includes/tetgen_pfem_refine_GLASS_CORRECT.h:
```diff
@@ -0,0 +1,1792 @@
+#ifdef _DEBUG
+ModelPart::NodesContainerType& ModelNodes = ThisModelPart.Nodes();
+				if( *(ModelNodes).find( outnew.tetrahedronlist[base]).base() == *(ThisModelPart.Nodes().end()).base() ) 
+					KRATOS_ERROR(std::logic_error,"trying to use an inexisting node","");
```
It won't compile in debug mode because the KRATOS_ERROR macro has changed to streaming form:
```C++
KRATOS_ERROR << "trying to use an inexisting node";
```
And for check only in debug mode you can use the following macro:
```C++
KRATOS_DEBUG_ERROR_IF(ModelNodes->find( outnew.tetrahedronlist[base]) == (ThisModelPart.Nodes().end()) ) << "trying to use an inexisting node";
```
avoiding the if defined macro.

pooyan-dadvand commented on applications/MeshingApplication/external_includes/trigen_glass_forming.h:
```diff
@@ -0,0 +1,1363 @@
+					pnode->X0() = pnode->X() - disp[0];
+					pnode->Y0() = pnode->Y() - disp[1];
+					pnode->Z0() = 0.0;
+					KRATOS_WATCH("Added node at the EDGE")
```
Seems to be a debugging watch...

pooyan-dadvand commented on applications/MeshingApplication/external_includes/trigen_glass_forming.h:
```diff
@@ -0,0 +1,1363 @@
+
+
+        clean_triangulateio(out_mid);
+        KRATOS_WATCH("ln420");
```
a debugging watch...

pooyan-dadvand commented on applications/MeshingApplication/external_includes/trigen_glass_forming.h:
```diff
@@ -0,0 +1,1363 @@
+        {
+            char mesh_regen_opts[] = "YJq1.4arn";
+            triangulate(mesh_regen_opts, &in2, &out2, &vorout2);
+            KRATOS_WATCH("Adaptive remeshing executed")
```
more watch

playpaolo replied:
I would like to keep this watch

pooyan-dadvand commented on applications/MeshingApplication/external_includes/trigen_glass_forming.h:
```diff
@@ -0,0 +1,1363 @@
+        typedef PointVector::iterator PointIterator;
+        typedef std::vector<double>               DistanceVector;
+        typedef std::vector<double>::iterator     DistanceIterator;
+        KRATOS_WATCH("ln449");
```
+1

pooyan-dadvand commented on applications/MeshingApplication/external_includes/trigen_glass_forming.h:
```diff
@@ -0,0 +1,1363 @@
+#ifdef _DEBUG
+            ModelPart::NodesContainerType& ModelNodes = ThisModelPart.Nodes();
+            if( *(ModelNodes).find( out2.trianglelist[base]).base() == *(ThisModelPart.Nodes().end()).base() )
+                KRATOS_THROW_ERROR(std::logic_error,"trying to use an inexisting node","");
```
Here the macro is correct but changing it to new format would give you more information about the error.

pooyan-dadvand commented on applications/MeshingApplication/external_includes/trigen_glass_forming.h:
```diff
@@ -0,0 +1,1363 @@
+        KRATOS_WATCH("ln749");
+
+        clean_triangulateio(in2);
+        KRATOS_WATCH("ln752");
```
watch....

pooyan-dadvand commented on applications/MeshingApplication/external_includes/trigen_glass_forming.h:
```diff
@@ -0,0 +1,1363 @@
+        clean_triangulateio(in2);
+        KRATOS_WATCH("ln752");
+        clean_triangulateio(out2);
+        KRATOS_WATCH("ln754");
```
more

pooyan-dadvand commented on applications/MeshingApplication/external_includes/trigen_glass_forming.h:
```diff
@@ -0,0 +1,1363 @@
+                    unsigned int counter = 0;
+                    for(PointIterator i=res.begin(); i!=res.begin() + n_points_in_radius ; i++)
+                    {
+                        if ( (*i)->FastGetSolutionStepValue(IS_BOUNDARY,1)==1.0 && res_distances[k] < 0.2*radius && res_distances[k] > 0.0 )
```
Not in this pull request but in the future it would be great to change the use of variable IS_BOUNDARY to the flag:

```c++
if((*i)->Is(BOUNDARY)  && res_distances[k]< 0.2*radius && res_distances[k] > 0.0 )
```

It is faster and uses less memory.

playpaolo replied:
Will do so.  Thank you!

pooyan-dadvand commented on applications/MeshingApplication/external_includes/trigen_glass_forming.h:
```diff
@@ -0,0 +1,1363 @@
+			pnode->FastGetSolutionStepValue(IS_BOUNDARY)=1.0;
+			pnode->FastGetSolutionStepValue(IS_FREE_SURFACE)=1.0;
+			pnode->FastGetSolutionStepValue(IS_FLUID)=1.0;
+			pnode->FastGetSolutionStepValue(IS_STRUCTURE)=0.0;
```
many of these variables can be changed to corresponding flag. (you can leave this for at the moment)

pooyan-dadvand commented on applications/PFEMapplication/CMakeLists.txt~:
```diff
@@ -0,0 +1,46 @@
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
```
you should add ~ files to your gitignore. This file should not be added to repository

pooyan-dadvand commented on applications/ULFapplication/CMakeLists.txt~:
```diff
@@ -0,0 +1,59 @@
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
```
To be removed from repository!

playpaolo replied:
done

pooyan-dadvand commented on applications/ULFapplication/test_exemples/slosh.gid/convergence_info.txt:
```diff
@@ -0,0 +1,451 @@
+Bulk-100.0
```
Do you need this to be in the repo?

playpaolo replied:
will add it to ignore now.

playpaolo replied:
removing

pooyan-dadvand commented on applications/ULFapplication/test_exemples/dam_2d.gid/dam_2d_0.230433928733.post.vv:
```diff
@@ -0,0 +1,96 @@
+BeginZE /home/pavel/Kratos/applications/ULFapplication/test_exemples/dam_2d.gid/dam_2d_0.230433928733.post.vv
```
I would add the vv files to gitignore....

playpaolo replied:
removed all the old test examples for the moment

pooyan-dadvand commented on applications/ULFapplication/custom_elements/ulf_axisym.h:
```diff
@@ -221,47 +221,45 @@ class UlfFrac3DSwimming
+
+    virtual void save(Serializer& rSerializer) const
+    {
+        KRATOS_SERIALIZE_SAVE_BASE_CLASS(rSerializer, Element);
```
It is important to save and load all your member variables in serializer

playpaolo replied:
I didnt understand this one. What am I missing in the serializer? Which variable?

pooyan-dadvand commented on applications/ULFapplication/custom_processes/add_wall_process.h:
```diff
@@ -0,0 +1,310 @@
+    //ModelPart& fluid_model_part, ModelPart& structure_model_part, ModelPart& combined_model_part)
+    //: mr_fluid_model_part(fluid_model_part), mr_structure_model_part(structure_model_part), mr_combined_model_part(combined_model_part)
+    {
+	KRATOS_WATCH(" INSIDE ADD WALL NODES CONSTRUCTOR") 
```
debugging watch....

pooyan-dadvand commented on applications/ULFapplication/custom_processes/add_wall_process.h:
```diff
@@ -0,0 +1,310 @@
+	av_mesh_size/=fluid_model_part.Nodes().size();
+
+
+	KRATOS_WATCH(av_mesh_size)
```
more watches....

pooyan-dadvand commented on applications/ULFapplication/custom_processes/pressure_calculate_process_axisym.h:
```diff
@@ -0,0 +1,397 @@
+        {
+            im->Calculate(PRESSURE,dummy,proc_info);
+        }
+        KRATOS_WATCH("Execute of Pressure Calulate Process");
```
+1

pooyan-dadvand commented on applications/ULFapplication/custom_processes/remove_and_save_wall_process.h:
```diff
@@ -0,0 +1,322 @@
+        KRATOS_TRY
+	ModelPart fluid_only_model_part;
+
+        fluid_only_model_part.Elements().clear();
```
You should use the Remove methods of modelpart instead of clearing directly the arrays. This will make the submodelparts in inconsistent situation!!

pooyan-dadvand commented on applications/ULFapplication/custom_processes/remove_and_save_wall_process.h:
```diff
@@ -0,0 +1,322 @@
+	fluid_model_part.Nodes()=fluid_only_model_part.Nodes();
+	fluid_model_part.Elements()=fluid_only_model_part.Elements();
+	fluid_model_part.Conditions()=fluid_only_model_part.Conditions();	
+
```
This method should be carefully rewritten!!

pooyan-dadvand commented on applications/ULFapplication/custom_utilities/assign_point_neumann_conditions.h:
```diff
@@ -86,45 +88,138 @@ namespace Kratos
+//KRATOS_WATCH(ThisModelPart.Conditions().size())
 			//KRATOS_WATCH("BEFORE SORTING")
 			ThisModelPart.Conditions().Sort();
+KRATOS_WATCH(ThisModelPart.Conditions().size())
```
watch...

pooyan-dadvand commented on applications/ULFapplication/custom_utilities/assign_point_neumann_conditions.h:
```diff
@@ -86,45 +88,138 @@ namespace Kratos
+			for(ModelPart::NodesContainerType::iterator it = ThisModelPart.NodesBegin(); 
+				it!=ThisModelPart.NodesEnd(); it++)
+			 {
+			 	Condition::NodesArrayType temp;	
```
it would be more efficient to create the array outside the loop and clear it in each iteration....

playpaolo replied:
corrected it as you requested

pooyan-dadvand commented on applications/ULFapplication/custom_utilities/assign_point_neumann_conditions.h:
```diff
@@ -86,45 +88,138 @@ namespace Kratos
+				  }
+
+			}
+KRATOS_WATCH(ThisModelPart.Conditions().size())
```
+1

pooyan-dadvand reviewed (changes requested):
I agree with styling issues mentioned by @loumalouomega and apart from that I would request changes for removing unnecessary files (like backup files) from the pull request and there are many debugging watches that can be cleaned. The rest of the comments are in the code and to encourage you for better styling and adapting to new changes in the core for future.

RiccardoRossi commented:
Panel,    is it intentional that you removed superlu?

Riccardo

On Oct 19, 2017 11:17 AM, "playpaolo" <notifications@github.com> wrote:

> Latest version of Updated Lagrangian Fluid models including the glass
> forming module
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/KratosMultiphysics/Kratos/pull/865
> Commit Summary
>
>    - Updating to macth new format
>    - Cleaning obsolete elements
>    - Updated the SetUpDofSet function
>    - Added elements and conditions and updated constructors
>    - python_scripts folder inside the application folder removed
>    - removed results files
>    - Updating (2to3 Py syntax) and adding files for glass simulation
>    - Fixing a bug in re-meshing with structure
>    - 2to3
>    - Nodal h at lonely nodes
>    - Meshing parameteres
>    - reform dof set at each iteration enabled
>    - minor change
>    - updated glass files
>    - gitignore
>    - merging master into feature-ulf
>    - added missing argument
>
> File Changes
>
>    - *M* .gitignore
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-0>
>    (12)
>    - *D* applications/ExternalSolversApplication/
>    custom_external_libraries/SuperLU_4.3/FORTRAN/c_fortran_dgssv.c.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-1>
>    (175)
>    - *D* applications/ExternalSolversApplication/
>    custom_external_libraries/SuperLU_4.3/SRC/cgsisx.c.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-2>
>    (723)
>    - *D* applications/ExternalSolversApplication/
>    custom_external_libraries/SuperLU_4.3/SRC/cgsitrf.c.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-3>
>    (629)
>    - *D* applications/ExternalSolversApplication/
>    custom_external_libraries/SuperLU_4.3/SRC/cgssvx.c.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-4>
>    (619)
>    - *D* applications/ExternalSolversApplication/
>    custom_external_libraries/SuperLU_4.3/SRC/clacon.c.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-5>
>    (221)
>    - *D* applications/ExternalSolversApplication/
>    custom_external_libraries/SuperLU_4.3/SRC/dgsisx.c.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-6>
>    (727)
>    - *D* applications/ExternalSolversApplication/
>    custom_external_libraries/SuperLU_4.3/SRC/dgsitrf.c.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-7>
>    (626)
>    - *D* applications/ExternalSolversApplication/
>    custom_external_libraries/SuperLU_4.3/SRC/dgssvx.c.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-8>
>    (614)
>    - *D* applications/ExternalSolversApplication/
>    custom_external_libraries/SuperLU_4.3/SRC/dlacon.c.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-9>
>    (236)
>    - *D* applications/ExternalSolversApplication/
>    custom_external_libraries/SuperLU_4.3/SRC/ilu_ccopy_to_ucol.c.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-10>
>    (202)
>    - *D* applications/ExternalSolversApplication/
>    custom_external_libraries/SuperLU_4.3/SRC/ilu_dcopy_to_ucol.c.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-11>
>    (199)
>    - *D* applications/ExternalSolversApplication/
>    custom_external_libraries/SuperLU_4.3/SRC/ilu_ddrop_row.c.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-12>
>    (311)
>    - *D* applications/ExternalSolversApplication/
>    custom_external_libraries/SuperLU_4.3/SRC/ilu_scopy_to_ucol.c.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-13>
>    (199)
>    - *D* applications/ExternalSolversApplication/
>    custom_external_libraries/SuperLU_4.3/SRC/ilu_zcopy_to_ucol.c.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-14>
>    (202)
>    - *D* applications/ExternalSolversApplication/
>    custom_external_libraries/SuperLU_4.3/SRC/mc64ad.f.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-15>
>    (2002)
>    - *D* applications/ExternalSolversApplication/
>    custom_external_libraries/SuperLU_4.3/SRC/sgsisx.c.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-16>
>    (723)
>    - *D* applications/ExternalSolversApplication/
>    custom_external_libraries/SuperLU_4.3/SRC/sgsitrf.c.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-17>
>    (626)
>    - *D* applications/ExternalSolversApplication/
>    custom_external_libraries/SuperLU_4.3/SRC/slacon.c.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-18>
>    (236)
>    - *D* applications/ExternalSolversApplication/
>    custom_external_libraries/SuperLU_4.3/SRC/zgsisx.c.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-19>
>    (723)
>    - *D* applications/ExternalSolversApplication/
>    custom_external_libraries/SuperLU_4.3/SRC/zgsitrf.c.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-20>
>    (629)
>    - *D* applications/ExternalSolversApplication/
>    custom_external_libraries/SuperLU_4.3/SRC/zlacon.c.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-21>
>    (221)
>    - *D* applications/ExternalSolversApplication/
>    custom_external_libraries/SuperLU_4.3/TESTING/MATGEN/Cnames.h.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-22>
>    (197)
>    - *M* applications/MeshingApplication/custom_
>    python/add_meshers_to_python.cpp
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-23>
>    (74)
>    - *A* applications/MeshingApplication/external_
>    includes/tetgen_pfem_refine_GLASS_CORRECT.h
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-24>
>    (1792)
>    - *A* applications/MeshingApplication/external_
>    includes/trigen_glass_forming.h
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-25>
>    (1363)
>    - *A* applications/PFEMapplication/CMakeLists.txt~
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-26>
>    (46)
>    - *D* applications/ThermoMechanicalApplication/
>    custom_elements/SUPG_conv_3d.cpp~
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-27>
>    (352)
>    - *D* applications/ThermoMechanicalApplication/
>    custom_elements/SUPG_conv_diff_2d.cpp~
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-28>
>    (391)
>    - *D* applications/ThermoMechanicalApplication/
>    custom_elements/SUPG_conv_diff_2d.h~
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-29>
>    (437)
>    - *D* applications/ThermoMechanicalApplication/
>    custom_elements/SUPG_conv_diff_3d.cpp~
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-30>
>    (421)
>    - *D* applications/ThermoMechanicalApplication/
>    custom_elements/heat_contact_2d.cpp~
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-31>
>    (226)
>    - *D* applications/ThermoMechanicalApplication/
>    custom_elements/heat_contact_2d.h~
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-32>
>    (311)
>    - *D* applications/ThermoMechanicalApplication/
>    custom_elements/heat_contact_3d.cpp~
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-33>
>    (226)
>    - *D* applications/ThermoMechanicalApplication/
>    custom_elements/heat_contact_3d.h~
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-34>
>    (311)
>    - *D* applications/ThermoMechanicalApplication/
>    custom_python/add_custom_processes_to_python.cpp~
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-35>
>    (87)
>    - *D* applications/ThermoMechanicalApplication/custom_python/thermo_
>    mechanical_python_application.cpp~
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-36>
>    (102)
>    - *M* applications/ULFapplication/CMakeLists.txt
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-37>
>    (12)
>    - *A* applications/ULFapplication/CMakeLists.txt~
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-38>
>    (59)
>    - *M* applications/ULFapplication/ULF_application.cpp
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-39>
>    (45)
>    - *M* applications/ULFapplication/ULF_application.h
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-40>
>    (19)
>    - *A* applications/ULFapplication/custom_conditions/Point_Neumann2D.cpp
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-41>
>    (311)
>    - *R* applications/ULFapplication/custom_conditions/Point_Neumann2D.h
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-42>
>    (140)
>    - *M* applications/ULFapplication/custom_conditions/Point_Neumann3D.cpp
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-43>
>    (33)
>    - *A* applications/ULFapplication/custom_conditions/Point_
>    Neumann_Axisym.cpp
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-44>
>    (314)
>    - *R* applications/ULFapplication/custom_conditions/Point_
>    Neumann_Axisym.h
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-45>
>    (143)
>    - *A* applications/ULFapplication/custom_elements/ulf_axisym.cpp
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-46>
>    (623)
>    - *R* applications/ULFapplication/custom_elements/ulf_axisym.h
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-47>
>    (92)
>    - *D* applications/ULFapplication/custom_elements/ulf_frac2d_
>    swimming.cpp
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-48>
>    (0)
>    - *M* applications/ULFapplication/custom_elements/ulf_frac2d_swimming.h
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-49>
>    (0)
>    - *M* applications/ULFapplication/custom_elements/ulf_frac3d_
>    swimming.cpp
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-50>
>    (0)
>    - *M* applications/ULFapplication/custom_processes/add_wall_process.h
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-51>
>    (0)
>    - *M* applications/ULFapplication/custom_processes/merge_model_
>    parts_process.h
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-52>
>    (0)
>    - *M* applications/ULFapplication/custom_processes/pressure_
>    calculate_process_axisym.h
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-53>
>    (0)
>    - *M* applications/ULFapplication/custom_processes/remove_and_
>    save_wall_process.h
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-54>
>    (0)
>    - *M* applications/ULFapplication/custom_python/add_custom_
>    utilities_to_python.cpp
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-55>
>    (0)
>    - *M* applications/ULFapplication/custom_python/add_processes_
>    to_python.cpp
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-56>
>    (0)
>    - *M* applications/ULFapplication/custom_python/kratos_ulf_
>    python_application.cpp
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-57>
>    (0)
>    - *M* applications/ULFapplication/custom_strategies/builder_and_
>    solvers/residualbased_elimination_quasiincompresible_builder_
>    and_solver.h
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-58>
>    (0)
>    - *M* applications/ULFapplication/custom_utilities/assign_point_
>    neumann_conditions.h
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-59>
>    (0)
>    - *M* applications/ULFapplication/custom_utilities/ulf_utilities.h
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-60>
>    (0)
>    - *M* applications/ULFapplication/python_scripts/ulf_PGLASS.py
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-61>
>    (0)
>    - *M* applications/ULFapplication/python_scripts/ulf_frac.py
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-62>
>    (0)
>    - *M* applications/ULFapplication/python_scripts/ulf_frac_strategy.py
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-63>
>    (0)
>    - *M* applications/ULFapplication/python_scripts/ulf_fsi.py
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-64>
>    (0)
>    - *M* applications/ULFapplication/python_scripts/ulf_fsi_inc.py
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-65>
>    (0)
>    - *M* applications/ULFapplication/python_scripts/ulf_strategy_PGLASS.py
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-66>
>    (0)
>    - *M* applications/ULFapplication/python_scripts/ulf_strategy_python.py
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-67>
>    (0)
>    - *M* applications/ULFapplication/python_scripts/ulf_strategy_
>    python_inc.py
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-68>
>    (0)
>    - *M* applications/ULFapplication/python_scripts/ulf_strategy_
>    python_inc_rigid.py
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-69>
>    (0)
>    - *M* applications/ULFapplication/test_exemples/dam_2d.gid/dam_2d.cond
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-70>
>    (0)
>    - *M* applications/ULFapplication/test_exemples/dam_2d.gid/dam_2d.elem
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-71>
>    (0)
>    - *M* applications/ULFapplication/test_exemples/dam_2d.gid/dam_2d.init
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-72>
>    (0)
>    - *M* applications/ULFapplication/test_exemples/dam_2d.gid/dam_2d.node
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-73>
>    (0)
>    - *M* applications/ULFapplication/test_exemples/dam_2d.gid/dam_2d.prop
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-74>
>    (0)
>    - *M* applications/ULFapplication/test_exemples/dam_2d.gid/dam_
>    2d_0.230433928733.post.png
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-75>
>    (0)
>    - *M* applications/ULFapplication/test_exemples/dam_2d.gid/dam_
>    2d_0.230433928733.post.vv
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-76>
>    (0)
>    - *M* applications/ULFapplication/test_exemples/dam_2d.gid/dam_
>    2d_out.mdpa
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-77>
>    (0)
>    - *M* applications/ULFapplication/test_exemples/slosh.gid/
>    convergence_info.txt
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-78>
>    (0)
>    - *M* applications/ULFapplication/test_exemples/slosh.gid/fluid_
>    ulf_var.py
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-79>
>    (0)
>    - *M* applications/ULFapplication/test_exemples/slosh.gid/slosh.py
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-80>
>    (0)
>    - *M* applications/ULFapplication/test_exemples/slosh.gid/slosh_
>    out.mdpa
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-81>
>    (0)
>    - *M* applications/convection_diffusion_application/python_
>    scripts/nonlinear_convection_diffusion_solver.py
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-82>
>    (0)
>    - *M* applications/freezing_soil_application/constitutive_laws/
>    clay_and_sand_model.cpp.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-83>
>    (0)
>    - *M* applications/trilinos_application/custom_python/
>    trilinos_python_application.cpp.bak
>    <https://github.com/KratosMultiphysics/Kratos/pull/865/files#diff-84>
>    (0)
>
> Patch Links:
>
>    - https://github.com/KratosMultiphysics/Kratos/pull/865.patch
>    - https://github.com/KratosMultiphysics/Kratos/pull/865.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/KratosMultiphysics/Kratos/pull/865>, or mute the
> thread
> <https://github.com/notifications/unsubscribe-auth/AHr7EXTYNxYMCzKJPxF8bQhI4-1eZd09ks5stxO9gaJpZM4P-7kI>
> .
>

playpaolo committed:
Removal of obsolete files

removed obsolete examples. Now the test_examples folder will be empty. Next I will create 2 test_examples considering the updated version of the application and corresp. interface

playpaolo committed:
Removing "unclean" and "trial" files

playpaolo committed:
removal of debugging "watches"

pooyan-dadvand commented:
I forgot to insist in my review that the changes in remove and save wall process are really important! The actual implementation cause serious problems in sub model parts.

playpaolo committed:
removed temporary

playpaolo committed:
Changes according to Pooyan's comments

in processes: changed operations on nodes and elements (model_part.Nodes().Clear)
to operations over model part: model_part.RemoveNodes().

pooyan-dadvand commented on applications/ULFapplication/custom_processes/add_wall_process.h:
```diff
@@ -0,0 +1,312 @@
+
+        fluid_model_part.Nodes().Sort();
+        fluid_model_part.Elements().Sort();
+        fluid_model_part.Conditions().Sort();
```
These sorts are not needed anymore

pooyan-dadvand commented on applications/ULFapplication/custom_processes/add_wall_process.h:
```diff
@@ -0,0 +1,312 @@
+	av_mesh_size/=fluid_model_part.Nodes().size();
+
+
+	KRATOS_WATCH(av_mesh_size)
```
Still there are watches...

pooyan-dadvand reviewed (changes requested):
There are still some watches and sorts to remove

pooyan-dadvand commented on applications/ULFapplication/custom_processes/remove_and_save_wall_process.h:
```diff
@@ -107,58 +107,62 @@ class RemoveAndSaveWallNodesProcess
+        //fluid_only_model_part.Elements().Sort();
+        //fluid_only_model_part.Conditions().Sort();
 
 	wall_model_part.Nodes().Sort();
```
This sort is not needed anymore.

pooyan-dadvand commented on applications/ULFapplication/custom_utilities/assign_point_neumann_conditions.h:
```diff
@@ -91,28 +91,25 @@ namespace Kratos
 		{			
 			KRATOS_TRY;
-			KRATOS_WATCH("INSide of AssignPointNeumannConditions UTILITY")
+			KRATOS_WATCH("Inside of AssignPointNeumannConditions UTILITY")
```
Do you need this watch?

pooyan-dadvand reviewed (changes requested):
minor changes for sorts and watches.

playpaolo committed:
Id re-numbering.

Corrected Id re-numbering after removing/adding nodes. 
Changed the function of remove_and_save so as not to haved a temporary fluid_only model part

playpaolo committed:
Clean-up

removed unused functions

playpaolo committed:
Clean-up

removed Sorts and unnecessary KRATOS_watches

pooyan-dadvand reviewed (approved):
Thanks for applying comments! 
:+1:

pooyan-dadvand commented:
@playpaolo do you want to add your application to release binary? if you like please follow the release issue #866

pooyan-dadvand commented:
note that the checks should pass before you can merge it.

roigcarlo commented:
Can we merge this?

pooyan-dadvand committed:
Merge branch 'master' into feature-ulf-app-update

pooyan-dadvand commented:
@playpaolo would you please merge it after checks are passed? I have removed the conflict.

pooyan-dadvand commented:
@playpaolo would you please merge this before it become outdated?
